### ISSUE TITLE
mode=refresh fails with spec: Forbidden: pod updates may not change fields other than `containers[*].image` or `spec.activeDeadlineSeconds`

### ISSUE THREAD

**Issue by sdodson** [openshift/origin-metrics#198]
I'm working on automated installation and when I detect that all the metrics components aren't running and we have a failed deployer pod I attempt to re-run the deployer with MODE=refresh. When I do that I get the following error.

OCP 3.3 environment.

```
[root@ose3-master ~]# oc logs -n openshift-infra metrics-deployer-78dp2
+ deployer_mode=refresh
+ image_prefix=brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/openshift3/
+ image_version=3.3.0
+ master_url=https://kubernetes.default.svc:443
+ [[ 3 == \/ ]]
++ parse_bool false REDEPLOY
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ redeploy=false
+ '[' false == true ']'
+ mode=refresh
+ '[' refresh = redeploy ']'
++ parse_bool false IGNORE_PREFLIGHT
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ ignore_preflight=false
+ cassandra_nodes=1
++ parse_bool true USE_PERSISTENT_STORAGE
++ local v=true
++ '[' true '!=' true -a true '!=' false ']'
++ echo true
+ use_persistent_storage=true
+ cassandra_pv_size=10Gi
+ metric_duration=7
+ user_write_access=false
+ heapster_node_id=nodename
+ metric_resolution=10s
+ project=openshift-infra
+ master_ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+ token_file=/var/run/secrets/kubernetes.io/serviceaccount/token
+ dir=/etc/deploy/_output
+ secret_dir=/secret
+ rm -rf /etc/deploy/_output
+ mkdir -p /etc/deploy/_output
+ chmod 700 /etc/deploy/_output
+ mkdir -p /secret
+ chmod 700 /secret
chmod: changing permissions of '/secret': Read-only file system
+ :
+ hawkular_metrics_hostname=hawkular-metrics.apps.example.com
+ hawkular_metrics_alias=hawkular-metrics
+ hawkular_cassandra_alias=hawkular-cassandra
++ date +%s
+ openshift admin ca create-signer-cert --key=/etc/deploy/_output/ca.key --cert=/etc/deploy/_output/ca.crt --serial=/etc/deploy/_output/ca.serial.txt --name=metrics-signer@1471275687
+ '[' -n 1 ']'
+ oc config set-cluster master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWV0cml
jcy1kZXBsb3llci10b2tlbi11anVseSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiNzFjOWQ1MTItNjBjZS0xMWU2LWJhZWYtNTI1NDAwOWNjMTlhIiwic3ViIjoic3lzdGVtOnNlcnZ
pY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.wWVIy4aDKXM6IlpBZFrzmli-tOagyEsM0Em9kqSaTSXT0VUMLV7laWiDkSJ7LXWaK9nJAyQevk0b5zzgHj68fXGu5CR7U5L_EBKo3dp-lPQsxvb10nCBsHeL2EjW686lviDyMD4toJeluGlUYXK_E_LIM4WxiRvJGUhJOuzVR_D5WszpeEipFQfBTn849Vb25cj2jR7DUBdd8fZB9Z3
TI8V1csMdL4adIEKqfxdOmeYg8nQ1bS01l1v7OD3fVebwj-yPvpUdrNhcV8hPw-uUBUUuzsRpzULM4EGlWWbF21Amf-hpe4tUGr0kDlr0qxPdmM6xo4LaJyObPTXlmPAKpA
user "account" set.
+ oc config set-context current --cluster=master --user=account --namespace=openshift-infra
context "current" set.
+ oc config use-context current
switched to context "current".
+ old_kc=/etc/deploy/.kubeconfig
+ KUBECONFIG=/etc/deploy/_output/kube.conf
+ '[' -z 1 ']'
+ oc config set-cluster deployer-master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "deployer-master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials deployer-account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIj
oibWV0cmljcy1kZXBsb3llci10b2tlbi11anVseSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiNzFjOWQ1MTItNjBjZS0xMWU2LWJhZWYtNTI1NDAwOWNjMTlhIiwic3ViIjoic3lzdG
VtOnNlcnZpY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.wWVIy4aDKXM6IlpBZFrzmli-tOagyEsM0Em9kqSaTSXT0VUMLV7laWiDkSJ7LXWaK9nJAyQevk0b5zzgHj68fXGu5CR7U5L_EBKo3dp-lPQsxvb10nCBsHeL2EjW686lviDyMD4toJeluGlUYXK_E_LIM4WxiRvJGUhJOuzVR_D5WszpeEipFQfBTn849Vb25cj2jR7DUB
dd8fZB9Z3TI8V1csMdL4adIEKqfxdOmeYg8nQ1bS01l1v7OD3fVebwj-yPvpUdrNhcV8hPw-uUBUUuzsRpzULM4EGlWWbF21Amf-hpe4tUGr0kDlr0qxPdmM6xo4LaJyObPTXlmPAKpA
user "deployer-account" set.
+ oc config set-context deployer-context --cluster=deployer-master --user=deployer-account --namespace=openshift-infra
context "deployer-context" set.
+ '[' -n 1 ']'
+ oc config use-context deployer-context
switched to context "deployer-context".
+ case $deployer_mode in
+ '[' false '!=' true ']'
+ validate_preflight
+ set -e
+ set +x

PREFLIGHT CHECK SUCCEEDED
validate_master_accessible: ok
validate_hostname: The HAWKULAR_METRICS_HOSTNAME value is deemed acceptable.
validate_deployer_secret:
Deleting any previous deployment (leaving route and PVCs)
POD_NAME metrics-deployer-78dp2
The Pod "metrics-deployer-78dp2" is invalid.
spec: Forbidden: pod updates may not change fields other than `containers[*].image` or `spec.activeDeadlineSeconds`
```

**Comment by mwringe** [comment]
if you are running on OCP, you will need to open a BZ for it. These issues are only for the origin metrics components. Otherwise the fix may not get into OCP until the next version.

**Comment by mwringe** [comment]
Closing as I cannot reproduce with origin.

The issue appears to be that 'oc label pod' is no longer functioning properly, which may be an issue with the version of oc included with the OCP 3.3 version.
